### ISSUE TITLE
[Bug] Qwen3-TTS: starting the torch profiler with with_stack deadlocks the tts_engine stage process at concurrency 16

### ISSUE THREAD

**Issue by db-ol** [sgl-project/sglang-omni#1779]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang-omni/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

Starting the torch profiler with SGLANG_TORCH_PROFILER_WITH_STACK=1 while a Qwen3-TTS server is busy at concurrency 16 can deadlock the tts_engine stage process.

The profiler start does not return, the scheduler stops issuing decode batches, GPU utilization drops to zero, in flight requests time out, and the server does not recover. The same profiling sequence at concurrency 1 completes every time.

It reproduces on clean main as well as with an unrelated local patch, and a second engineer on the team hit it independently on their own setup. py-spy shows a GIL self deadlock inside PyTorch's PythonTracer, the with_stack machinery, so the defect appears to be in the torch profiler rather than in sglang-omni logic. sglang-omni exposes the affected path through /start_profile and the SGLANG_TORCH_PROFILER_WITH_STACK switch.

Status: reproduced, root caused, workaround available (with_stack off), upstream fix pending.

**Stacks of the hung stage process** (py-spy dump, two dumps 20 s apart show the same frames)

Main thread of the tts_engine stage process, still inside the profiler start three minutes after /start_profile returned 200 to the client:

```
Thread "MainThread"
    pthread_cond_timedwait (libc.so.6)
    pybind11::gil_scoped_acquire::gil_scoped_acquire (libtorch_python.so)
    torch::detail::(anonymous namespace)::ConcretePyInterpreterVTable::incref (libtorch_python.so)
    c10::TensorImpl::incref_pyobject (libc10.so)
    torch::profiler::impl::RawTensorMetadata::RawTensorMetadata (libtorch_cpu.so)
    torch::profiler::impl::(anonymous namespace)::toTensorMetadata (libtorch_python.so)
    torch::profiler::impl::(anonymous namespace)::PythonTracer::recordPyCall (libtorch_python.so)
    torch::profiler::impl::(anonymous namespace)::PythonTracer::PythonTracer (libtorch_python.so)
    torch::profiler::impl::(anonymous namespace)::getTracer (libtorch_python.so)
    torch::profiler::impl::python_tracer::PythonTracerBase::make (libtorch_cpu.so)
    torch::profiler::impl::RecordQueue::RecordQueue (libtorch_cpu.so)
    torch::autograd::profiler::enableProfiler (libtorch_cpu.so)
    _start_trace (torch/autograd/profiler.py:398)
    start_trace (torch/profiler/profiler.py:236)
    start (torch/profiler/profiler.py:872)
    start (sglang_omni/profiler/torch_profiler.py:123)
    _on_profiler_start (sglang_omni/pipeline/stage/runtime.py:1797)
    _handle_message (sglang_omni/pipeline/stage/runtime.py:381)
```

The scheduler thread, waiting to take the GIL back after a released op:

```
Thread "scheduler-tts_engine"
    pthread_cond_timedwait (libc.so.6)
    pybind11::gil_scoped_release::~gil_scoped_release (libtorch_python.so)
    torch::autograd::THPVariable_contiguous (libtorch_python.so)
    forward_extend (flashattention_backend.py:1420)
    ...
    _forward_with_input_embeds (sglang_omni/models/qwen3_tts/model_runner.py:391)
    _run_batch (sglang_omni/scheduling/omni_scheduler.py:1408)
    _event_loop_normal (sglang_omni/scheduling/omni_scheduler.py:2342)
```

The vocoder followup worker is inside a layer_norm in chunked_decode, the request builders are inside the speech tokenizer forward, all waiting on the GIL. About 420 Python threads of the process sit in futex_wait (/proc/<pid>/task/*/wchan), no thread is active, nothing runs on the GPU.

**Reading**

Checked against torch v2.11.0 torch/csrc/autograd/profiler_python.cpp. The PythonTracer constructor takes the GIL through gil_and_restore_thread, then for every interpreter thread calls PyThreadState_Swap(thread_state) and replays that thread's existing frames through recordPyCall. When one of those frames is an nn.Module __call__, ValueCache::store records the module's parameters through toTensorMetadata and RawTensorMetadata, which increfs each parameter's Python object and takes a pybind11 gil_scoped_acquire. At that moment the thread state has been swapped to the thread being recorded, so pybind11 sees a thread state that is not the one bound to the calling OS thread, concludes the GIL is not held, and blocks in PyEval_AcquireThread on a lock its own thread already holds. Every other Python thread then queues behind the GIL.

This is why concurrency matters: the hang needs some other thread to be inside a module forward at the instant the profiler starts. At concurrency 16 the vocoder workers, the request builders and the scheduler are almost always mid forward, at concurrency 1 they rarely are. It also explains why the same concurrency 16 window sometimes passes.

**Workaround and suggestion**

Run the server with SGLANG_TORCH_PROFILER_WITH_STACK=0. With it off the same concurrency 16 window completed 3 of 3 times and exported its trace, with it on 3 of 5 runs deadlocked. The trace then has no Python stacks, but CUPTI kernel and runtime events are intact.

Suggestion for sglang-omni: default with_stack to off for serving processes, or document that with_stack is only safe on an idle server, and have _on_profiler_start log a warning when with_stack is requested while the scheduler has running requests. The torch side deserves an upstream report with the stack above, I can file it.

### Reproduction

Model: Qwen/Qwen3-TTS-12Hz-1.7B-Base, single process topology (examples/configs/qwen3_tts_1_7b.yaml).

```
SGLANG_TORCH_PROFILER_DIR=/tmp/prof SGLANG_TORCH_PROFILER_WITH_STACK=1 \
  sgl-omni serve --config examples/configs/qwen3_tts_1_7b.yaml --port 18083 --mem-fraction-static 0.5 \
  --tts_engine.engine.max_running_requests 32 --tts_engine.engine.max_queued_requests 32
```

1. Wait for /health.
2. Run the streaming SeedTTS benchmark at concurrency 1 (40 samples):
   `python -m benchmarks.eval.benchmark_tts_seedtts --generate-only --use-existing-server --stream --meta zhaochenyang20/seed-tts-eval-arrow --lang en --model Qwen/Qwen3-TTS-12Hz-1.7B-Base --port 18083 --max-samples 40 --warmup 4 --concurrency 1`
   and during it POST /start_profile {"run_id": "c1", "enable_torch": true}, wait 4 s, POST /stop_profile {"run_id": "c1"}. This window completes and the trace is exported.
3. Run the same benchmark at concurrency 16 (640 samples), and 25 s in POST /start_profile {"run_id": "c16", "enable_torch": true}, wait 4 s, POST /stop_profile {"run_id": "c16"}.
4. The server log stops showing decode batches right after "Starting End-to-End Torch profiler (run_id=c16)". No "Trace exported" line ever follows, the benchmark's requests hit the client timeout and the coordinator logs "Coordinator aborted req=...". GPU utilization stays at 0 percent with the server process alive.

Rates: 3 of 5 runs of this sequence hung (two on clean main 1452d08, one with an unrelated local patch on c2d193f). The control with SGLANG_TORCH_PROFILER_WITH_STACK=0 completed 3 of 3 with both traces exported. One run that profiled concurrency 16 without a prior concurrency 1 window did not hang. The second reporter's commit and settings will be added in a comment.

### Environment

- sglang-omni main c2d193f and 1452d08
- torch 2.11.0+cu130, triton 3.6.0, Python 3.12, one H200 (CUDA driver 13.2)
- py-spy 0.4.2 in a container started with --cap-add SYS_PTRACE for the dumps
